# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/android/nsWindow.cpp
# Commit: a2f0dae16c37
# Full Hash: a2f0dae16c3730b690d3268a7ca8db59cd4681a0
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2020-11-11 09:55:06
# Regressor Bug: 1674694
# File Overlap Count: 1
# Description:
#   Bug 1674694 - Use ConvertAPZHandledResult nsEventStatus_eConsumeNoDefault case. r=botond,geckoview-reviewers,agi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96332
# ==============================================================================

diff -r 5e275c31e9f4 -r a2f0dae16c37 widget/android/nsWindow.cpp
--- a/widget/android/nsWindow.cpp	Wed Nov 11 00:33:20 2020 +0000
+++ b/widget/android/nsWindow.cpp	Wed Nov 11 00:33:28 2020 +0000
@@ -360,13 +360,9 @@
         WheelDeltaAdjustmentStrategy::eNone);
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
-    int32_t ret =
-        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
-            ? INPUT_RESULT_HANDLED
-            : INPUT_RESULT_HANDLED_CONTENT;
-
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
-      return ret;
+      MOZ_ASSERT(result.mHandledResult, "Should have a valid APZHandledResult");
+      return ConvertAPZHandledResult(result.mHandledResult.value());
     }
 
     PostInputEvent([input, result](nsWindow* window) {
@@ -378,7 +374,9 @@
       case nsEventStatus_eIgnore:
         return INPUT_RESULT_UNHANDLED;
       case nsEventStatus_eConsumeDoDefault:
-        return ret;
+        return (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+                   ? INPUT_RESULT_HANDLED
+                   : INPUT_RESULT_HANDLED_CONTENT;
       default:
         MOZ_ASSERT_UNREACHABLE("Unexpected nsEventStatus");
         return INPUT_RESULT_UNHANDLED;
@@ -503,13 +501,9 @@
         nsWindow::GetEventTimeStamp(aTime), nsWindow::GetModifiers(aMetaState));
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
-    int32_t ret =
-        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
-            ? INPUT_RESULT_HANDLED
-            : INPUT_RESULT_HANDLED_CONTENT;
-
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
-      return ret;
+      MOZ_ASSERT(result.mHandledResult, "Should have a valid APZHandledResult");
+      return ConvertAPZHandledResult(result.mHandledResult.value());
     }
 
     PostInputEvent([input, result](nsWindow* window) {
@@ -521,7 +515,9 @@
       case nsEventStatus_eIgnore:
         return INPUT_RESULT_UNHANDLED;
       case nsEventStatus_eConsumeDoDefault:
-        return ret;
+        return (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+                   ? INPUT_RESULT_HANDLED
+                   : INPUT_RESULT_HANDLED_CONTENT;
       default:
         MOZ_ASSERT_UNREACHABLE("Unexpected nsEventStatus");
         return INPUT_RESULT_UNHANDLED;
@@ -704,14 +700,12 @@
     }
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
-    int32_t handled =
-        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
-            ? INPUT_RESULT_HANDLED
-            : INPUT_RESULT_HANDLED_CONTENT;
-
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
       if (returnResult) {
-        returnResult->Complete(java::sdk::Integer::ValueOf(handled));
+        MOZ_ASSERT(result.mHandledResult,
+                   "Should have a valid APZHandledResult");
+        returnResult->Complete(java::sdk::Integer::ValueOf(
+            ConvertAPZHandledResult(result.mHandledResult.value())));
       }
       return;
     }
@@ -737,7 +731,10 @@
               java::sdk::Integer::ValueOf(INPUT_RESULT_UNHANDLED));
           break;
         case nsEventStatus_eConsumeDoDefault:
-          returnResult->Complete(java::sdk::Integer::ValueOf(handled));
+          returnResult->Complete(java::sdk::Integer::ValueOf(
+              (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+                  ? INPUT_RESULT_HANDLED
+                  : INPUT_RESULT_HANDLED_CONTENT));
           break;
         default:
           MOZ_ASSERT_UNREACHABLE("Unexpected nsEventStatus");
